src/components/MediaGallery/MediaGallery.js

```diff
@@ -69,42 +66,28 @@ class MediaGallery extends Component {
     uiStore: PropTypes.object
   }
 
-  constructor(props) {
-    super(props);
+  static defaultProps = {
+    mediaItems: []
+  };
 
-    this.featuredMedia = Array.isArray(props.mediaItems) && props.mediaItems[0];
-  }
+  state = { featuredMediaIndex: 0 };
 
-  @observable _featuredMedia
-
-  @computed get featuredMedia() {
-    return this._featuredMedia;
-  }
-
-  set featuredMedia(value) {
-    this._featuredMedia = value;
-  }
-
-  /**
-   * Media item click handler
-   * @param {SyntheticEvent} event - Event
-   * @param {Object} mediaItem - Media item object
-   * @returns {undefined}
-   */
-  @action handleMediaItemClick = (event, mediaItem) => {
-    this.featuredMedia = mediaItem;
+  getClickHandler(index) {
+    return () => {
```

Comment:
With the snippets

```js
@action handleMediaItemClick = (event, mediaItem) => {}
```

```jsx
<MediaItem mediaItem={mediaItem} onClick={this.handleMediaItemClick} />
```
The `onClick` callback of MediaItem component passes the `mediaItem` object as the 2nd option in the callback. You can use that to determine which one was clicked when dealing when multiple items in an array. It can be any prop, you could use index instead if that's the desire.

The same functionality could be abstractd to a HOC.

---

You could also use data attributes on the element with the `onClick` callback and in your handler, use `event.targert.dataset.index`, though it's basically the same thing as above but messier.

---

I think we should strive to be as efficient as possible while we have the ability to do so. 


The eslint rule we use for reference: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md.
And the example: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md#protips

Reply:
Thanks, I didn't notice that MediaItem accepted and passed back an `index` prop.